### PR TITLE
kitty: install completions

### PR DESCRIPTION
**Pull request by jlesquembre**
###### Motivation for this change
Install completions for kitty

**NOTE:** I've only tested `fish` completions.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by Ma27** [comment]
First of all thanks a lot for this!

Bash completion works perfectly fine, but I only got ZSH working after running `source /run/current-system/sw/share/zsh/site-functions/_kitty` in the test VM.

Do you have an idea how to get ZSH working as well? Unless I can do some further digging to get ZSH completions working.

**Comment by jlesquembre** [comment]
I don't know why doesn't work, I'm not a ZSH user, sorry. To add more information, I just copied the way auto-completions are implemented for `autojump` and `fd`:
- https://github.com/NixOS/nixpkgs/blob/42bdc36a84c6a6cadd70c963d172d322e48c1db0/pkgs/tools/misc/fd/default.nix#L20-L23
- https://github.com/NixOS/nixpkgs/blob/42bdc36a84c6a6cadd70c963d172d322e48c1db0/pkgs/tools/misc/autojump/default.nix#L20-L23

I assumed that completions are working for that packages, if here ZSH completions are broken, maybe are also broken for that packages?

File: pkgs/applications/misc/kitty/default.nix [hunk 1]
```diff
@@ -45,6 +45,11 @@ buildPythonApplication rec {
     cp -r linux-package/{bin,share,lib} $out
     wrapProgram "$out/bin/kitty" --prefix PATH : "$out/bin:${stdenv.lib.makeBinPath [ imagemagick xsel ]}"
     runHook postInstall
+
+    mkdir -p "$out/share/"{bash-completion/completions,fish/vendor_completions.d,zsh/site-functions}
+    "$out/bin/kitty" + complete setup fish > "$out/share/fish/vendor_completions.d/kitty.fish"
+    "$out/bin/kitty" + complete setup bash > "$out/share/bash-completion/completions/kitty.bash"
+    "$out/bin/kitty" + complete setup zsh > "$out/share/zsh/site-functions/_kitty"
```
Comment (Mic92) [inline pkgs/applications/misc/kitty/default.nix]: It only works when explicitly sourcing the file, but not by putting into fpath:

https://github.com/kovidgoyal/kitty/blob/8ceb941051b89b7c50850778634f0b6137aa5e6e/docs/index.rst#zsh

Just skip zsh for now I would say.

Comment (Ma27) [reply pkgs/applications/misc/kitty/default.nix]: yeah let's drop that for now. If somebody needs ZSH completions, it's still possible to source them in `programs.zsh.shellInit`.

Comment (Ma27) [reply pkgs/applications/misc/kitty/default.nix]: I updated the install script and left a comment

**Review by Ma27 — APPROVED** [review]

**Comment by Ma27** [comment]
@jlesquembre thanks!

File: pkgs/applications/misc/kitty/default.nix [hunk 2]
```diff
@@ -45,6 +45,12 @@ buildPythonApplication rec {
     cp -r linux-package/{bin,share,lib} $out
     wrapProgram "$out/bin/kitty" --prefix PATH : "$out/bin:${stdenv.lib.makeBinPath [ imagemagick xsel ]}"
     runHook postInstall
+
+    # ZSH completions need to be invoked with `source`:
+    # https://github.com/kovidgoyal/kitty/blob/8ceb941051b89b7c50850778634f0b6137aa5e6e/docs/index.rst#zsh
+    mkdir -p "$out/share/"{bash-completion/completions,fish/vendor_completions.d,zsh/site-functions}
```
Comment (Mic92) [inline pkgs/applications/misc/kitty/default.nix]: nitpick:

```diff
- mkdir -p "$out/share/"{bash-completion/completions,fish/vendor_completions.d,zsh/site-functions}
+ mkdir -p "$out/share/"{bash-completion/completions,fish/vendor_completions.d}
```

**Comment by wedens** [comment]
@Ma27 zsh completions will probably work if you add `environment.pathsToLink = [ "/share/zsh" ];`

**Comment by Ma27** [comment]
I doubt it. I tested this with `programs.zsh.enable = true` which already links `/share/zsh` which is also why I could run `source /run/current-system/sw/share/zsh/site-functions/_kitty` to get the completions working.

**Pull request merged**